RichEditorView/Classes/RichEditorView.swift

```diff
@@ -439,6 +445,75 @@ extension RichEditorView: UIGestureRecognizerDelegate {
 // MARK: - Utilities
 extension RichEditorView {
     
+    /** Sets content offset for scrollView based on caret location **/
+    
+    private func fixScrollView(editor: RichEditorView) {
+        
+        var scrollView = editor.webView.scrollView
+        
+        var contentHeight: CGFloat?
+        let htmlHeight = editor.runJS("document.getElementById('editor').clientHeight;")
+        if let n = NSNumberFormatter().numberFromString(htmlHeight) {
+            let floatValue = CGFloat(n)
+            contentHeight = floatValue
+        } else {
+            contentHeight = scrollView.frame.height
+        }
+        scrollView.contentSize = CGSizeMake(scrollView.frame.width, contentHeight!)
+        
+        /* Caret above scrollView bounds */
+        
+        // Usually getCaretPosition() returns the amount the caret is offset from the current contentOffset, so "-40" would
+        // indicate that the contentOffset would need to be decreased by 40.
+        // However, when the cursor is on a new line with no html content, getCaretPosition() returns the absolute position
+        // of the caret.  The resulting code differentiates between caret position values greater than 0 that could either mean
+        // the caret position is beyond the scrollView lower bounds, or above the scrollView upper bounds and on a new line.
+        // The code also checks if the caret position is within the scrollView bounds to avoid scrolling when caret is 
+        // not on the first line of the scrollView.
+        
+        let data = convertStringToDictionary(editor.getCaretPosition())
+        
+        // Checks if caret is on new line with no content
+        let newLine = data!["newLine"] as! Bool
+        if let caretPositionNumeric = data!["height"] as? NSNumber {
+            
+            let caretFloat = CGFloat(caretPositionNumeric) - scrollView.contentOffset.y
+            
+            var CGFloatCaretPositionNumeric = CGFloat(caretPositionNumeric.floatValue)
+            if newLine {
+                if (caretFloat + 24.0) > (scrollView.bounds.size.height) {
```

Comment:
@cjwirth  If the user can change the font size, then I need to add a new method that gets the current caret height, and uses that value instead of the hardcoded `24.0`.